### DISCUSSION TITLE
Performance measurements

### DISCUSSION THREAD

**Discussion by brandonpollack23** [rimbu-org/rimbu#135] [General]
So I want to start by saying thank you so much. 

I’ve been thinking of peeking up from my mostly OS and systems world and having some fun with typescript, but I really wanted clojure collections with types. 

This so scratches that itch. 

I’m curious about performance measurement. Has anything been done yet?

**Comment by vitoke** [comment]
Thanks for the kind words.

I started some performance measurements some time ago, I only managed to write tests for the `List` data structure. You can find that project in the `performance` folder in the root. I see that currently it doesn't run out of the box anymore but it is easy to fix. I'll do that soon.

In general, I'm not dissatisfied with the performance results so far. In creating the library, I've always kept performance in mind.

Let me know if there is anything in particular that you would like to see or know..

**Comment by brandonpollack23** [comment]
I have continued my self update on front end and Solid captured my
interest.

I’m going to see if it makes sense to use this library along with that or
not (seems like they like to have internal mutability).

If I can I’ll see about measuring something with both this and native
mutable structures.

**Comment by vitoke** [comment]
Yes Solid looks nice, though just as with React hooks, I'm not really happy with functions that have some internal magic to do state. I prefer a functional way to do Input props -> Output view, and keep the state/logic separate from that. Haven't yet found the holy grail but curious to hear your findings, and whether it's possible to combine Rimbu into Solid. I think it should just work, as long you stay away from `immer` because I tried that with Redux ToolKit and that completely didn't work. Which is why I don't really like `immer`..

**Comment by brandonpollack23** [comment]
So I know it is definitely somewhat possible with their store api and can
be a little more efficient with their reconcile call which adds a key for
deep diff comparison of large state trees.

I’m just an amplifier though because I don’t fully understand that yet. I
wonder if we can ask Ryan?
